fetch.bs

```diff
@@ -4070,9 +4070,38 @@ Range Requests</cite>. [[HTTP-RANGE]] However, this is not widely supported by b
 
    <li><p>If the <i>CORS flag</i> is set, <var>httpRequest</var>'s <a for=request>method</a> is
    neither `<code>GET</code>` nor `<code>HEAD</code>`, or <var>httpRequest</var>'s
-   <a for=request>mode</a> is "<code>websocket</code>", then <a for="header list">append</a>
-   `<code>Origin</code>`/the result of <a>serializing a request origin</a> with
-   <var>httpRequest</var>, to <var>httpRequest</var>'s <a for=request>header list</a>.
+   <a for=request>mode</a> is "<code>websocket</code>", then:
+   <ol>
+    <li><p>Let <var>origin</var> be the result of <a>serializing a request origin</a> with
+    <var>httpRequest</var>.
+    <li><p>If <var>request</var>'s <i>CORS flag</i> is unset, then switch on
```

Comment:
I suspect we also don't want to apply these steps when mode is "websocket" as that is effectively CORS too. And at that point I wonder if we can restructure this a bit. Perhaps something like:

1. Let serializedOrigin be the result of serializing a request origin with ...
2. If the CORS flag is set or httpRequest's mode is "websocket", then append Origin/serializedOrigin to httpRequest's header list.
3. Otherwise, if httpRequest's method is neither GET nor HEAD:
   1. The Referrer Policy sanitizing origin steps...

What do you think?